### ISSUE TITLE
Offline Circulation Tool (KOCT) -Firefoxin lisäosan varmenneongelma 

### ISSUE THREAD

**Issue by pirliis** [KohaSuomi/Koha#1547]
### Mikä vikana?

Kun Firefoxin lisäosassa määrittelee yhteystiedot ja tämän jälkeen tekee configuraatio testin, tulee ilmoitus:
![image](https://github.com/user-attachments/assets/83a14b29-6081-442b-a921-b5f5b1c334fd)

Jos valitsee "OK" tai "Älä lähetä varmennetta", molemmilla valinnoilla ohjelma antaa ilmoituksen ->"Configuration error:403 Forbidden"

Yhteisön bugitiketti:  https://bugs.koha-community.org/bugzilla3/show_bug.cgi?id=37060

**Comment by pirliis** [comment]
Yhteisön tiketissä on kommentoitu, että toimisi KOCTin 0.4.21 ja Kohan versiolla 24.05.08. Testasin myös OUTIn testillä ko. KOCT-versiolla ja meidän Koha-versiolla 24.05.01, mutta ei toiminut. Jos korjaus on Kohan 24.05.08 versiossa, voiko sitä tuoda meille erikseen vai odotammeko seuraavaan versiopäivitykseen?

**Comment by johannaraisa** [comment]
Kaivellaan ja kokeillaan, kyllähän sen tähän meidän versioon voi lisätä.

**Comment by AnneliO** [comment]
Jos ymmärsin oikein, niin vaatisi tiketin https://bugs.koha-community.org/bugzilla3/show_bug.cgi?id=38826 korjauksen myös, jotta liitännäinen toimisi oikein.

Asensin liitännäisen uuden version ja testailin sitä outi-testillä ja sain tuon saman forbidden, mitä muutkin olivat bugissa https://bugs.koha-community.org/bugzilla3/show_bug.cgi?id=37060 raportoineet saaneensa liitännäisen päivityksen jälkeen.

**Comment by AnneliO** [comment]
Testailin outi-nextillä KOCT-liitännäistä ja se tuntui ainakin mulla toimivan. Mites muilla?

**Comment by pknuuti** [comment]
En saa käyttöön KOCTia Vaaran nextillä. Ei tule kuvaketta esiin. Työpöytäsovelluksen yhteydettömän tilan lainaus onnistuu kuten pitääkin.

**Comment by AnneliO** [comment]
KOCT:n asennuslinkki löytyy Lainaus ja palautus -sivun alaosasta:

<img width="1280" height="666" alt="Image" src="https://github.com/user-attachments/assets/4fd0d1bf-7a36-479e-8bbe-e392f81b860b" />

Kun selaimen lisäosan on asentanut, sen avauskuvake pitäisi näkyä selaimen oikeassa yläkulmassa selaimen valikon lähellä:

<img width="231" height="103" alt="Image" src="https://github.com/user-attachments/assets/1995b249-4357-44c7-adc4-75ac48931457" />

Kun lisäosan avaa, pitää ensimmäisenä käydä asetuksissa asettamassa nextin osoite, käyttäjätunnus ja salasana ja testaamassa asetukset Test configuration -napista (pitäisi tulla vihreällä viereen teksti Configuration ok). Valitse valikosta vielä kirjasto ja alareunasta Apply directly to Koha. Sitten vielä Save.

**Comment by pknuuti** [comment]
Sepä siinä olikin se ongelma, kun tuo kuvake ei tule selaimen yläkulmaan. Ei vaikka useamman kerran yritin (tallensi lisäosan - poistin lisäosan - tallensin uudelleen - päivitin...) Onkohan meillä estetty näiden lisäosien ottaminen käyttöön, se kävisi selityksestä.

**Comment by AnneliO** [comment]
Näkyykö kuvake, jos avaat valikosta Laajennukset ja teemat -osion? Siellä näkyy ainakin mulla listattuna asennetut lisäosat.

<img width="1341" height="540" alt="Image" src="https://github.com/user-attachments/assets/0b52a5da-d1f7-4086-8e0a-3fae044b083d" />

**Comment by pknuuti** [comment]
Näkyy se tuolla valikossa, mutta ei selaimessa kuten pitäisi. Mikähän sen piilottaa?

<img width="1171" height="426" alt="Image" src="https://github.com/user-attachments/assets/0f8b434a-5539-4e0a-9332-1f5e2380d03d" />

**Comment by pknuuti** [comment]
Sain tehtyä tuolla laajennusten puolella asetukset, mutta edelleenkään ei näy kuvake tuolla selaimessa kuten pitäisi. Onpa merkillistä.

**Comment by AnneliO** [comment]
Koitapa klikata tuossa valikon ja kuvakkeiden vasemmalla puolella tyhjässä kohdassa hiiren vasemmalla painikkeella, jolloin avautuu pieni valikko.
<img width="231" height="103" alt="Image" src="https://github.com/user-attachments/assets/1995b249-4357-44c7-adc4-75ac48931457" />

Valitse avautuvasta valikosta "Muokkaa työkalupalkkia"
<img width="403" height="173" alt="Image" src="https://github.com/user-attachments/assets/8c8f74d6-45df-4661-80cb-b748d85cc284" />

Näkyykö avautuvalla sivulla KOCT:n kuvaketta?

<img width="1016" height="334" alt="Image" src="https://github.com/user-attachments/assets/9ae08611-6f54-4413-9679-e37396c03497" />

Jos näkyy, niin vedä se valikkoriville valikon vasemmalle puolelle hiirellä.

**Comment by pknuuti** [comment]
Ei näy KOCT-kuvaketta tuossa näkymässä.
Oliskohan tällä jotain tekemistä asian kanssa: "Tietoja tästä laajennuksesta
This addon is a tool for librarian. It allows users to checkin and checkout items while offline and then send these transactions to Koha.

Available in Czech, English, French, German, Italian and Spanish.

This addon does not work in private browsing mode."
Mä kyllä laitoin ruksin sallimaan yksityisen selauksen näkymässä.

**Comment by AnneliO** [comment]
Se voi tosiaan olla, että se ei nyt vaan ala toimimaan yksityisen selauksen kanssa. 🤔

**Comment by AnneliO** [comment]
Joo, mullakin kuvake katoaa näkyvistä, kun avaan yksityisen selauksen ikkunan.

**Comment by pirliis** [comment]
> Testailin outi-nextillä KOCT-liitännäistä ja se tuntui ainakin mulla toimivan. Mites muilla?

OUTI-nextillä toimi myös minulla. 😊

**Comment by AnneliO** [comment]
Pystytkö @pknuuti testaamaan muuten kuin yksityisen selauksen ikkunassa?

**Comment by pknuuti** [comment]
Nyt sain testattua KOCTin toimivaksi nextillä. Selaimeni ei tosin toimi ihan samalla tavalla (Firefox 128.8.0esr), kuvake ei tule suoraan tuonne osoiteriville, vaan vähän piiloon. En tiedä mikä sen aiheuttaa. Olennaista on, että lainauksen pystyy tekemään KOCTilla ja tuomaan sen Kohaan onnistuneesti.

<img width="398" height="236" alt="Image" src="https://github.com/user-attachments/assets/52c1c4c8-5aa0-45f7-92ec-e38d9e7107f4" />

**Comment by AnneliO** [comment]
Tämän tiketin voinee sulkea, koska varmenneongelma poistuu versionvaihdon yhteydessä. :)

**Comment by pirliis** [comment]
Joo, suljen tiketin, kun ainakin Vaaran ja OUTIn testauksissa toimi.
